QGL/ChannelLibraries.py

```diff
@@ -423,7 +423,12 @@ def new_APS(self, label, address, **kwargs):
 
     @check_for_duplicates
     def new_TDM(self, label, address, **kwargs):
-        return Channels.Processor(label=label, model="TDM", address=address, trigger_interval=250e-6)
+        chans = []
+        for k in range(7): # TDM has 7 digital inputs
+            chans.append(Channels.DigitalInput(label=f"DigitalInput-{label}-{k}", channel=k, channel_db=self.channelDatabase))
+        tdm = Channels.Processor(label=label, model="TDM", address=address, trigger_interval=250e-6, channels=chans, channel_db=self.channelDatabase)
+        self.add_and_update_dict(tdm)
```

Comment:
Out of curiosity, shouldn't this TDM have been added to the dict before? Was that a bug?

Reply:
It was added, but without any channels